src/internal_events/mod.rs

```diff
@@ -210,6 +210,39 @@ pub fn emit(event: impl InternalEvent) {
     event.emit_metrics();
 }
 
+// Possible values for the "error_type" tag
+pub enum ErrorTypes {
+    FieldMissing,
+    InvalidMetric,
+    MappingFailed,
+    MatchFailed,
+    ParseFailed,
+    RenderError,
+    SerializationFailed,
+    TargetFieldExists,
+    TemplateError,
+    TypeConversionFailed,
+    ValueInvalid,
+}
+
+impl ErrorTypes {
+    fn to_str(&self) -> &str {
+        match self {
+            ErrorTypes::FieldMissing => "field_missing",
+            ErrorTypes::InvalidMetric => "invalid_metric",
+            ErrorTypes::MappingFailed => "mapping_failed",
+            ErrorTypes::MatchFailed => "match_failed",
+            ErrorTypes::ParseFailed => "parse_failed",
+            ErrorTypes::RenderError => "render_error",
+            ErrorTypes::SerializationFailed => "serialization_failed",
+            ErrorTypes::TargetFieldExists => "target_field_exists",
+            ErrorTypes::TemplateError => "template_error",
+            ErrorTypes::TypeConversionFailed => "type_conversion_failed",
+            ErrorTypes::ValueInvalid => "value_invalid",
+        }
+    }
+}
+
```

Comment:
I'm not sure this type of structure has any real benefits over a set of constants, at least without further work on emitting these errors. It doesn't enforce (in the compiler) that we only use these values, and requires extra code at each invocation (which admittedly will be completely inlined in release builds). It does add a namespace (`ErrorTypes`) but that can also be provided with a module.